### PR TITLE
Implement conversion traits for PyBytes

### LINKED ISSUE #373: bytes should translate to &[u8]
Per a discussion in the gitter chat, it seems that currently the proper traits are not implemented to allow for Python's `bytes` objects to be automatically translated into `&[u8]` parameters. Right now, in order to write a function that takes `bytes`, you need to use `&PyBytes`:

```rust
#[pyfunction]
fn print_bytes(_py: Python, x: &PyBytes) {
    println!("{:?}", x.as_bytes())
}
```

If you instead use `&[u8]` directly, there's an error at compile time because the `pyo3::PyTypeInfo` trait is not implemented for `[u8]`. I think it should be possible to pass both a `bytes` and `bytearray` object into a Rust function that takes `&[u8]`.

I am using version 0.6.0-alpha.4, and with this code:

```rust
#[pyfunction]
fn print_bytes(_py: Python, x: &[u8]) {
    println!("{:?}", x.as_bytes())
}
```

I get this error message:

```
error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
  --> src/lib.rs:36:1
   |
36 | #[pyfunction]
   | ^^^^^^^^^^^^^ doesn't have a size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `[u8]`
   = note: to learn more, visit <...>
   = note: required because of the requirements on the impl of
           `pyo3::FromPyObject<'_>` for `&[u8]`

error[E0277]: the trait bound `[u8]: pyo3::PyTypeInfo` is not satisfied
  --> src/lib.rs:36:1
   |
36 | #[pyfunction]
   | ^^^^^^^^^^^^^ the trait `pyo3::PyTypeInfo` is not implemented for `[u8]`
   |
   = note: required because of the requirements on the impl
           of `pyo3::PyTryFrom<'_>` for `[u8]`
   = note: required because of the requirements on the impl
           of `pyo3::FromPyObject<'_>` for `&[u8]`

error: aborting due to 2 previous errors
```

CC @thedrow

### PR DESCRIPTION
**Pull request by Alexander-N**
Closes #373

### CODE REVIEW

File: tests/test_bytes.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+use pyo3::prelude::*;
+use pyo3::types::PyBytes;
+use pyo3::wrap_pyfunction;
+
+mod common;
+
+#[pyfunction]
+fn get_pybytes() -> PyObject {
```
Comment (kngwyu) [inline tests/test_bytes.rs]: Since we recommend users to return `&Py~` types(in this case, `&PyBytes`), I don't think we need this test.

**Comment by kngwyu** [comment]
Thanks.

File: tests/test_bytes.rs (hunk 1 above)
Comment (Alexander-N) [reply tests/test_bytes.rs]: Done

Comment (Alexander-N) [reply tests/test_bytes.rs]: Actually, it is not clear to me how to return `&PyBytes` directly, see https://gitter.im/PyO3/Lobby?at=5d83e8d456677e767a225079

Comment (kngwyu) [reply tests/test_bytes.rs]:
```rust
#[pyclass]
struct ByteSequence {
    elements: Vec<u8>,
}

#[pymethods]
impl ByteSequence {
    fn bytes<'py>(&'py self, py: Python<'py>) -> &'py PyBytes {
        PyBytes::new(py, &self.elements)
    }
}
```

Comment (Alexander-N) [reply tests/test_bytes.rs]: I see, thanks. Should this be added as a test?

Comment (Alexander-N) [reply tests/test_bytes.rs]: Ok, I saw it already is.

**Pull request merged**